### PR TITLE
feat: change test runner to jest

### PR DESCRIPTION
**Pull request by kevinccbsg**
### Type:
* [ ] cicd: helm, docker & cicd adjustments.
* [X] feature: new capabilities.
* [ ] fix: bug, hotfix, etc.
* [X] refactor: enhacements.
* [ ] style: changes in styles.
* [ ] other: docs, tests.

### What's the focus of this PR:
This PR adds jest as a test runner. I will wait to merge this first #10

### CODE REVIEW

File: generators/app/templates/root/_package.json [hunk 1]
```diff
@@ -21,13 +20,12 @@
...
+    "eslint-plugin-jest": "^23.8.2",
     "husky": "^4.2.3",
+    "jest": "^25.4.0",
+    "jest-config": "^25.4.0",
+    "jest-junit": "^10.0.0",
     "make-manifest": "^1.0.1",
-    "mocha": "^5.2.0",
-    "mocha-junit-reporter": "^1.18.0",
-    "nyc": "^14.1.1",
     "sinon": "^7.3.2",
```
Comment (w3dani) [inline generators/app/templates/root/_package.json:29]: You can also remove sinon

File: generators/app/templates/root/_package.json [hunk 2]
```diff
@@ -6,9 +6,8 @@
...
     "local": "SERVICE_ENV=local node index.js",
     "start": "node index.js",
-    "test-ci": "SERVICE_ENV=test mocha --reporter mocha-junit-reporter",
-    "test:report": "SERVICE_ENV=test nyc --reporter=lcovonly mocha --reporter mocha-junit-reporter './test/**/*.test.js'",
-    "test": "SERVICE_ENV=test mocha --reporter spec './test/**/*.test.js'",
+    "test:report": "cross-env NODE_ENV=test jest --detectOpenHandles --forceExit --coverage",
+    "test": "cross-env NODE_ENV=test jest --detectOpenHandles --forceExit --reporters=default",
     "lint": "eslint .",
     "manifest": "node_modules/make-manifest/bin/make-manifest",
     "qa": "npm run lint && npm run test",
```
Comment (w3dani) [inline generators/app/templates/root/_package.json:13]: are we using this step?

File: generators/app/templates/root/_package.json (hunk 1 above)
Comment (kevinccbsg) [reply generators/app/templates/root/_package.json:29]: Good catch Thanks!!

File: generators/app/templates/root/_package.json (hunk 2 above)
Comment (kevinccbsg) [reply generators/app/templates/root/_package.json:13]: I do not think so, I'll remove it

**Comment by kevinccbsg** [comment]
@w3dani the lastest commit includes PR feedback 👍

**Pull request merged**